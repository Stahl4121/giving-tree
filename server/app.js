// Stripe Server

const app = require('express')();
const bodyParser = require('body-parser');
const stripe = require('stripe')('sk_test_avCwuzIvg45JxkjItTyqRGH600JEoryzyP');
const admin = require('firebase-admin');
const serviceAccount = require("./secret/givingtree-cfs-firebase-adminsdk-grrpd-764871e22c.json");

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  databaseURL: "https://givingtree-cfs.firebaseio.com"
});

app.use(bodyParser.text());
app.use(bodyParser.urlencoded({ extended: true }));

app.post('/charge', async (req, res) => {
  let source = req.body.split(' amount: ')[0];
  let amount = req.body.split(' amount: ')[1].split(' description: ')[0];
  let description = req.body.split(' amount: ')[1].split(' description: ')[1].split(' account: ')[0];
  let account = req.body.split(' amount: ')[1].split(' description: ')[1].split(' account: ')[1];
  try {
    let { status } = await stripe.charges.create({
      amount: amount,
      currency: 'usd',
      description: description,
      source: source,
    }, {
      stripeAccount: account,
    }).then(function (charge) {
      // asynchronously called
      return charge;
    });

    res.json({ status });
  } catch (err) {
    console.log(err);
    res.status(500).end();
  }
});

app.post('/create', async (req, res) => {
  try {
    let connected_account_id = await stripe.oauth.token({
      grant_type: 'authorization_code',
      code: req.body,
    }).then(function (response) {
      // asynchronously called
      return response.stripe_user_id;
    });
    return res.json({ 'stripeId': connected_account_id });

  } catch (err) {
    res.status(500).end;
  }
})



//Get current user's ID token
//If metaadmin, call "getUserByEmail" w/ the community foundation email, update status to accepted/rejected
//If not metaadmin, get current user's ID token and set custom claim 'cf' to true
app.post('/setCustomClaims', (req, res) => {

  // Get the ID token passed.
  const idToken = req.body.idToken;
  const cfEmail = req.body.cfEmail;
  const cfStatus = req.body.cfStatus;

  // Verify the ID token and decode its payload.
  admin.auth().verifyIdToken(idToken).then((claims) => {
    console.log('verified token\n');

    //Check if metaadmin
    if (typeof claims.admin !== 'undefined' && claims.admin) {
      //Check that cfEmail and cfStatus were provided in request
      if (typeof cfEmail !== 'undefined' && typeof cfStatus !== 'undefined'
        && cfEmail !== '' && cfStatus !== '' && typeof claims.cfId !== 'undefined' ) {
        //Get the CF User and update their permissions
        admin.auth().getUserByEmail(cfEmail).then((user) => {
          if (cfStatus === 'current') {
            admin.auth().setCustomUserClaims(user.uid, { cfId: claims.cfId, status: 'current' })
              .then(function () {
                // Tell client to refresh token on user.
                res.end(JSON.stringify({ status: 'success' }));
              })
              .catch((error) => {
                res.end(JSON.stringify({ status: error }));
              });
          }
          else if (cfStatus === 'denied') {
              admin.auth().setCustomUserClaims(user.uid, { cfId: claims.cfId, status: 'denied' })
                .then(function () {
                  // Tell client to refresh token on user.
                  res.end(JSON.stringify({ status: 'success' }));
                })
                .catch((error) => {
                  res.end(JSON.stringify({ status: error }));
                });
          }
        })
          .catch((error) => {
            res.end(JSON.stringify({ status: error }));
          });
      }
    }
    else {
      console.log('not metaadmin\n');
      //Not metaadmin, so we this post must be made from CF Account Request
      //Get and store the cf document record id in the claims object
      admin.auth().getUser(claims.sub)
        .then((user) => {
          console.log('got user\n');
          admin.firestore().collection('communityFoundations').where('personal_email', '==', user.email)
            .get()
            .then(function (querySnapshot) {
              querySnapshot.forEach(function (doc) {
              console.log('got cf!\n' );
                admin.auth().setCustomUserClaims(claims.sub, { cfId: doc.id, status: 'requested' })
                  .then(() => {
                  console.log('set cc!\n');
                    // Tell client to refresh token on user.
                    res.end(JSON.stringify({ status: 'success' }));
                  })
                  .catch((error) => {
                  console.log('failed cc!\n');

                    res.end(JSON.stringify({ status: error }));
                  });
                  console.log('after cc\n');
              });
            })
            .catch((error) => {
              res.end(JSON.stringify({ status: error }));
            });
        })
        .catch((error) => {
          res.end(JSON.stringify({ status: error }));
        });
        console.log('end of not metaadmin\n');
    }
  })
  .catch((error) => {
    console.log('NOOO at the end\n');
    res.end(JSON.stringify({ status: error }));
  });
});

app.listen(process.env.PORT);
// app.listen(9000, () => console.log('Listening on port 9000')); 	