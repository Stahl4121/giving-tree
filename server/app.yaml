#TODO: Figure out what else should go in here
runtime: nodejs10
