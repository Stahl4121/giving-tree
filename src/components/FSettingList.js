import React, { useEffect, Component } from 'react';
import { Link } from 'react-router-dom';

import { Button, Typography, Container, Grid} from '@material-ui/core';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import ImageIcon from '@material-ui/icons/Image';
import TitleIcon from '@material-ui/icons/Title'
import PhoneIcon from '@material-ui/icons/Phone';
import HttpIcon from '@material-ui/icons/Http';
import EmailIcon from '@material-ui/icons/Email';
import ContactsIcon from '@material-ui/icons/Contacts';
import ContactsOutlinedIcon from '@material-ui/icons/ContactsOutlined';
import PhoneIphoneIcon from '@material-ui/icons/PhoneIphone';
import AlternateEmailIcon from '@material-ui/icons/AlternateEmail';

import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

export default function NonEditableData(props){
	return (
		<Container maxWidth="lg">
			<Grid container direction={'column'} style={{ 'margin-bottom': 15 }}>
				<Typography variant="h4">
					{props.cfInfo.name}
				</Typography>
				<Grid container direction='row'>
					<Button 
						color='primary'
						onClick={props.toggleEdit} 
						variant='outlined'
						style={{ 'margin': 2 }}
					>
						Edit
					</Button>
					<Link to={{ pathname: "/forgot" }}>
						<Button 
							variant='outlined'
							style={{ 'margin': 2}}
						>
							Change Password
						</Button>
					</Link>
					<ToggleActive 
						active={props.cfInfo.status}
						toggleAccountActive={props.toggleAccountActive}
					/>
				</Grid>
			</Grid>
			
			<Grid container direction={'row'}>
				<Grid item xs="6">
					<Typography variant="h5">Public Information</Typography>
					<Typography variant="p" paragraph={true}>This information will be displayed on all published giving opportunities.</Typography>
					<Grid container direction={'column'} spacing={2}>
						<StaticItem 
							icon={<TitleIcon />}
							label={"Foundation Name"}
							value={props.cfInfo.name}
						/>
						<StaticItem 
							icon={<EmailIcon />}
							label={"Public Email"}
							value={props.cfInfo.public_email}
						/>
						<StaticItem 
							icon={<PhoneIcon />}
							label={"Public Phone"}
							value={props.cfInfo.public_phone}
						/>
						<StaticItem 
							icon={<HttpIcon />}
							label={"Foundation URL"}
							value={props.cfInfo.foundation_url}
						/>
					</Grid>
				</Grid>
				
				<Grid item maxWidth="6">
					<Typography variant="h5">Personal Contact Information</Typography>
					<Typography variant="p" paragraph={true}>This information will be used to sign into and manage your account.</Typography>
					<Grid container direction={'column'} spacing={2}>
						<StaticItem 
							icon={<ContactsIcon />}
							label={"Contact First Name"}
							value={props.cfInfo.fname_contact}
						/>
						<StaticItem 
							icon={<ContactsOutlinedIcon />}
							label={"Contact Last Name"}
							value={props.cfInfo.lname_contact}
						/>
						<StaticItem 
							icon={<AlternateEmailIcon />}
							label={"Contact Email"}
							value={props.cfInfo.personal_email}
						/>
						<StaticItem 
							icon={<PhoneIphoneIcon />}
							label={"Contact Phone"}
							value={props.cfInfo.personal_phone}
						/>
					</Grid>
				</Grid>
			</Grid>
		</Container>
		
	);
}


function StaticItem(props) {
	return(
		<Grid container item spacing={2}>
			<Grid item>
				<Avatar>
					{React.cloneElement(props.icon)}
				</Avatar>
			</Grid>
			<Grid item sm container>
				<ListItemText 
					primary={props.label}
					secondary={props.value} 
				/>
			</Grid>
		</Grid>
	);
}


function ToggleActive(props) {
	console.log(props.active);
	const [open, setOpen] = React.useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleYesClose = () => {
		props.toggleAccountActive();
		setOpen(false);
	};
	const handleNoClose = () => {
    setOpen(false);
	};
	
	if(props.active === 'current'){
		return(
			<div>
				<Button 
					color='secondary'
					onClick={handleClickOpen} 
					variant='contained'
					style={{ 'margin': 2 }}
				>Deactivate Account</Button>
				<Dialog
					open={open}
					onClose={handleNoClose}
					aria-labelledby="alert-dialog-title"
					aria-describedby="alert-dialog-description"
				>
					<DialogTitle id="alert-dialog-title">{"Use Google's location service?"}</DialogTitle>
					<DialogContent>
						<DialogContentText id="alert-dialog-description">
							Are you sure you would like to deactivate your account?
						</DialogContentText>
					</DialogContent>
					<DialogActions>
						<Button onClick={handleNoClose} color="primary">
							No
						</Button>
						<Button onClick={handleYesClose} color="secondary" variant='contained' autoFocus>
							Yes
						</Button>
					</DialogActions>
				</Dialog>
			</div>
		);
	}else if(props.active === 'deactivated'){
		return(
			<div>
				<Button 
					color='primary'
					onClick={handleClickOpen} 
					variant='contained'
					style={{ 'margin': 2 }}
				>Activate Account</Button>
				<Dialog
					open={open}
					onClose={handleNoClose}
					aria-labelledby="alert-dialog-title"
					aria-describedby="alert-dialog-description"
				>
					<DialogTitle id="alert-dialog-title">{"Use Google's location service?"}</DialogTitle>
					<DialogContent>
						<DialogContentText id="alert-dialog-description">
							Are you sure you would like to activate your account?
						</DialogContentText>
					</DialogContent>
					<DialogActions>
						<Button onClick={handleNoClose} color="secondary">
							No
						</Button>
						<Button onClick={handleYesClose} color="primary" variant='contained' autoFocus>
							Yes
						</Button>
					</DialogActions>
				</Dialog>
			</div>
		);
	}
	else{
		return(null);
	}
}