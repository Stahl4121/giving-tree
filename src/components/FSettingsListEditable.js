import React, { useEffect } from 'react';

import { Button, Typography, Container, TextField, Grid } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import ImageIcon from '@material-ui/icons/Image';
import TitleIcon from '@material-ui/icons/Title'
import PhoneIcon from '@material-ui/icons/Phone';
import HttpIcon from '@material-ui/icons/Http';
import EmailIcon from '@material-ui/icons/Email';
import ContactsIcon from '@material-ui/icons/Contacts';
import ContactsOutlinedIcon from '@material-ui/icons/ContactsOutlined';
import PhoneIphoneIcon from '@material-ui/icons/PhoneIphone';
import AlternateEmailIcon from '@material-ui/icons/AlternateEmail';

import * as helper from '../helpers/ValidationHelper.js';

export default function EditableData(props){
	
	const [helperText, setHelperText] = React.useState({
    name: '',
    public_email: '',
    public_phone: '',
    foundation_url: '',
    fname_contact: '',
    lname_contact: '',
    personal_email: '',
    personal_phone: ''
	});
	
	const [changedText, setChangedText] = React.useState({});

	const onChange = (event, field) => {
		const value = event.target.value;
		setHelperText({ ...helperText, [field]: helper.validateField(field, value) },
			setChangedText({ ...changedText, [field]: value })
		);
	}

	const submitForm = () => {
		var canSubmit = true;
		for(const key in helperText){
			if(helperText[key] !== ''){
				canSubmit = false;
			}
		}
		if(canSubmit){
			props.onSubmit(changedText)
		}else{
			console.log("Dang it we can't, here's why: ");
			console.log(helperText)
		}
	}

	return (
		<Container maxWidth="lg">
		<Grid container direction={'column'} style={{ 'margin-bottom': 15 }}>
			<Typography variant="h4">
				{props.cfInfo.name}
			</Typography>
			<Grid container direction='row'>
				<Button 
					color='primary'
					onClick={submitForm} 
					variant='outlined'
					style={{ 'margin': 2 }}
				>
					Save
				</Button>
				<Button 
					onClick={props.toggleEdit} 
					variant='outlined'
					style={{ 'margin': 2 }}
				>
					Discard
				</Button>
			</Grid>
		</Grid>

			<Grid container direction={'row'}>
				<Grid item xs="6">
					<Typography variant="h5">Public Information</Typography>
					<Typography variant="p" paragraph={true}>This information will be displayed on all published giving opportunities.</Typography>
					<Grid container direction={'column'} spacing={3}>
						<EditItem 
							icon={<TitleIcon />}
							label={'Foundation Name'}
							default={props.cfInfo.name}
							change={(e, f) => onChange(e, 'name')}
							helper={helperText.name}
						/>
						<EditItem 
							icon={<EmailIcon />}
							label={'Public Email'}
							default={props.cfInfo.public_email}
							change={(e, f) => onChange(e, 'public_email')}
							helper={helperText.public_email}
						/>
						<EditItem 
							icon={<PhoneIcon />}
							label={'Public Phone'}
							default={props.cfInfo.public_phone}
							change={(e, f) => onChange(e, 'public_phone')}
							helper={helperText.public_phone}
						/>
						<EditItem 
							icon={<HttpIcon />}
							label={'Foundation URL'}
							default={props.cfInfo.foundation_url}
							change={(e, f) => onChange(e, 'foundation_url')}
							helper={helperText.foundation_url}
						/>
					</Grid>
				</Grid>

				<Grid item xs="6">
					<Typography variant="h5">Personal Contact Information</Typography>
					<Typography variant="p" paragraph={true}>This information will be used to sign into and manage your account.</Typography>
					<Grid container direction={'column'} spacing={3}>
						<EditItem 
							icon={<ContactsIcon />}
							label={'Contact First Name'}
							default={props.cfInfo.fname_contact}
							change={(e, f) => onChange(e, 'fname_contact')}
							helper={helperText.fname_contact}
						/>
						<EditItem 
							icon={<ContactsOutlinedIcon />}
							label={'Contact Last Name'}
							default={props.cfInfo.lname_contact}
							change={(e, f) => onChange(e, 'lname_contact')}
							helper={helperText.lname_contact}
						/>
						<EditItem 
							icon={<AlternateEmailIcon />}
							label={'Contact Email'}
							default={props.cfInfo.personal_email}
							helper={""}
							disabled={true}
						/>
						<EditItem 
							icon={<PhoneIphoneIcon />}
							label={'Contact Phone'}
							default={props.cfInfo.personal_phone}
							change={(e, f) => onChange(e, 'personal_phone')}
							helper={helperText.personal_phone}
						/>
					</Grid>
				</Grid>
			</Grid>
		</Container>
	);
}

function EditItem(props, disabled=false) {
	return(
		<Grid container item spacing={2}>
			<Grid item>
				<Avatar>
					{React.cloneElement(props.icon)}
				</Avatar>
			</Grid>
			<Grid item xs={8} sm container>
				<TextField 
					label={props.label} 
					defaultValue={props.default} 
					onChange={props.change} 
					helperText={props.helper} 
					disabled={props.disabled}
					error={props.helper !== ""}
					fullWidth
				/>
			</Grid>
		</Grid>
	);
}