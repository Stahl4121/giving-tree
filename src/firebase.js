//Reasoning for this file: https://stackoverflow.com/questions/48492047/where-do-i-initialize-firebase-app-in-react-application

// Firebase App (the core Firebase SDK) is always required and
// must be listed before other Firebase SDKs
import * as firebase from "firebase/app";

// Add the Firebase services that you want to use
import "firebase/auth";
import "firebase/firestore";

//Import everything, not recommended for production apps
//import firebase from "firebase";

const firebaseConfig = {
  apiKey: "AIzaSyAkeCK6w-qbvHzaOdEyg_PtpGQbElAv0_M",
  authDomain: "givingtree-cfs.firebaseapp.com",
  databaseURL: "https://givingtree-cfs.firebaseio.com",
  projectId: "givingtree-cfs",
  storageBucket: "givingtree-cfs.appspot.com",
  messagingSenderId: "709659064913",
  appId: "1:709659064913:web:d9decf22beb442f2e79670",
  measurementId: "G-KZ6PNG9K64"
};


firebase.initializeApp(firebaseConfig);

export default !firebase.apps.length ? firebase.initializeApp(firebaseConfig) : firebase.app();