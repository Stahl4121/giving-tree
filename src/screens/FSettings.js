import React, { useEffect } from 'react';
import AuthUserContext from '../auth/context.js';

import {firestore as FIRESTORE} from "firebase/app";
import firebase from '../firebase.js';

import EditableData from '../components/FSettingsListEditable.js'
import NonEditableData from '../components/FSettingList.js'


export default function FSettings() {
 
  // Set tab title
  useEffect(() => { document.title = 'Settings- Giving Tree'; }, []);
  var db = firebase.firestore();

  const [isEdit, setEdit] = React.useState(false);
  const user = React.useContext(AuthUserContext);

  const [getData, setGetData] = React.useState(true);

  // Foundation info
  const [cfInfo, setCfInfo] = React.useState({
    name: '',
    public_email: '',
    public_phone: '',
    foundation_url: '',
    fname_contact: '',
    lname_contact: '',
    personal_email: '',
    personal_phone: '',
    status: '',
    date_requested: '',
    date_approved: '',
    date_denied: '',
    date_deactivated: '',
  });

  
  function toggleEdit()
  {
    if(isEdit){
      setEdit(false);
    }else{
      setEdit(true);
    }
  }

  React.useEffect(() => {
    if(getData){
      if((user !== null)){
        FSettingsMethods().loadData(user, db, (doc) => {
          setCfInfo(doc.data());
          setGetData(false);
        });
      }
    }
  }, [getData, user, db]);
  
  if(isEdit){
    return(
      <EditableData
        isEdit={isEdit}
        cfInfo={cfInfo}
        toggleEdit={toggleEdit}
        onSubmit={(changedText) => FSettingsMethods().onSubmit(cfInfo, changedText, user, db, () => {
          setGetData(true);
          setEdit(false);
        })}
        functionLoadData={() => FSettingsMethods().loadData}
      />
    );
  }else{
    return(
      <NonEditableData
        cfInfo={cfInfo}
        toggleEdit={toggleEdit}
        onSubmit={(changedText) => FSettingsMethods().onSubmit(cfInfo, changedText, user, db, () => {
          setGetData(true, () => setEdit(false));
        })}
        toggleAccountActive={() => FSettingsMethods().toggleAccountActive(db, user, cfInfo, () => {
          setGetData(true);
        })}
      />
    );
  }
} 


// All this crapola is outside the original functional component for testing purposes.
// They say you should write your tests for your code, but since when do "they" know what they're talking about? *sobs*
export const FSettingsMethods = () => {
  const loadData = (user, db, callback) =>{
    // Foundation query
    db.collection('communityFoundations').where('personal_email','==', user.email)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          // doc.data() is never undefined for query doc snapshots
          //console.log(doc.id, " => ", doc.data());
          callback(doc);
        });
      })
      .catch(function(error) {
        console.log("Error getting documents: ", error);
      });
  }
  
  
  const onSubmit = (cfInfo, changedText, user, db, callback) =>{
    var temp = {...cfInfo};
    for(const key in changedText){
      temp = { ...temp, [key]: changedText[key] };
    }
  
    db.collection('communityFoundations').where('personal_email','==',user.email)
    .get()
    .then(function(querySnapshot) {
      querySnapshot.forEach(function(doc) {
        //console.log(doc.id, " => ", doc.data());
        // Build doc ref from doc.id
        db.collection("communityFoundations").doc(doc.id).update({
          name: temp.name,
          public_email: temp.public_email,
          public_phone: temp.public_phone,
          foundation_url: temp.foundation_url,
          fname_contact: temp.fname_contact,
          lname_contact: temp.lname_contact,
          personal_email: temp.personal_email,
          personal_phone: temp.personal_phone
        }).then(
          callback(temp)
        );
        console.log("Document successfully written!");
      });
    })
    .catch(function(error) {
      console.error("Error writing document: ", error);
    });
  }
  
  const toggleAccountActive = (db, user, cfInfo, callback) => {
    if(user){
      db.collection('communityFoundations').where('personal_email','==',user.email)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          const time = (cfInfo.date_deactivated === '') ? FIRESTORE.FieldValue.serverTimestamp() : '';
          db.collection("communityFoundations").doc(doc.id).update(
            {date_deactivated: time},
          ).then(
            () => callback(time)
          );
          console.log("Deactivation successfully toggled!")
        });
      })
      .catch(function(error) {
        console.error("Error writing document: ", error);
      });
    }
  }

  return {
    loadData,
    onSubmit,
    toggleAccountActive,
  };
}


