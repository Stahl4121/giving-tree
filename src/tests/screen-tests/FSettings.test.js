import React from 'react';
import { shallow, mount } from 'enzyme';
import AuthUserContext from '../../auth/context.js';

import FSettings from '../../screens/FSettings.js';
import {firestore as FIRESTORE} from "firebase/app";
import { StaticRouter } from 'react-router-dom';
import { FSettingsMethods } from '../../screens/FSettings.js';
import { EditableData, EditItem } from '../../components/FSettingsListEditable.js';
import { NonEditableData, ToggleActive } from '../../components/FSettingList.js';
import { ExpansionPanelActions } from '@material-ui/core';


const { mockFirebase } = require('firestore-jest-mock');
const { mockCollection, mockWhere } = require('firestore-jest-mock/mocks/firestore');

describe('FSettings', () => {

  mockFirebase({
    database: {
      communityFoundations: [
        {
          name: 'The Test Foundation',
          public_email: 'testy@tester.tested',
          public_phone: '1112223333',
          foundation_url: 'testme.com',
          fname_contact: 'Tabaath',
          lname_contact: 'Ben Test',
          personal_email: 'blah@yada.net',
          personal_phone: '9998887777',
          status: 'current',
          date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
          date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
          date_denied: '',
          date_deactivated: '',
        },
      ],
    },
  });
  const user = {email : 'blah@yada.net'};

  let firebase;
  let db;
  let wrap;

  beforeEach(() => {
    firebase = require('firebase');
    db = firebase.firestore();
    wrap = shallow(
      <AuthUserContext.Provider value={{user:{email: 'blah@yada.net'}}}>
        <StaticRouter>
          <FSettings />
        </StaticRouter>
      </AuthUserContext.Provider>
    )
  });

  it('Renders correctly', () => {  
    expect(wrap).toMatchSnapshot();
  });

  it('Loads correct data from the DB', () => {  
    
    FSettingsMethods().loadData(user, db, (doc) => expect(doc.data()).toEqual({
      name: 'The Test Foundation',
      public_email: 'testy@tester.tested',
      public_phone: '1112223333',
      foundation_url: 'testme.com',
      fname_contact: 'Tabaath',
      lname_contact: 'Ben Test',
      personal_email: 'blah@yada.net',
      personal_phone: '9998887777',
      status: 'current',
      date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
      date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
      date_denied: '',
      date_deactivated: '',
    }));
    
  });

  it('Submits correct data to the DB', () => {  
    var cfInfo = {
      name: 'The Test Foundation',
      public_email: 'testy@tester.tested',
      public_phone: '1112223333',
      foundation_url: 'testme.com',
      fname_contact: 'Tabaath',
      lname_contact: 'Ben Test',
      personal_email: 'blah@yada.net',
      personal_phone: '9998887777',
      status: 'current',
      date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
      date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
      date_denied: '',
      date_deactivated: '',
    };
    var changedText = {
      public_email: 'testie@tester.tested',
      public_phone: '9876543210',
    };

    FSettingsMethods().onSubmit(cfInfo, changedText ,user, db, (temp) => expect(temp).toEqual({
      name: 'The Test Foundation',
      public_email: 'testie@tester.tested',
      public_phone: '9876543210',
      foundation_url: 'testme.com',
      fname_contact: 'Tabaath',
      lname_contact: 'Ben Test',
      personal_email: 'blah@yada.net',
      personal_phone: '9998887777',
      status: 'current',
      date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
      date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
      date_denied: '',
      date_deactivated: '',
    }));
    
  });

  it('Submits correct time to thingy', () => {  
    var cfInfo = {
      name: 'The Test Foundation',
      public_email: 'testy@tester.tested',
      public_phone: '1112223333',
      foundation_url: 'testme.com',
      fname_contact: 'Tabaath',
      lname_contact: 'Ben Test',
      personal_email: 'blah@yada.net',
      personal_phone: '9998887777',
      status: 'current',
      date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
      date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
      date_denied: '',
      date_deactivated: '',
    };
    FSettingsMethods().toggleAccountActive(db, user, cfInfo , (time) => expect(time).toEqual(FIRESTORE.FieldValue.serverTimestamp()));
    
  });

  it('Submits correct time to thingy', () => {  
    var cfInfo = {
      name: 'The Test Foundation',
      public_email: 'testy@tester.tested',
      public_phone: '1112223333',
      foundation_url: 'testme.com',
      fname_contact: 'Tabaath',
      lname_contact: 'Ben Test',
      personal_email: 'blah@yada.net',
      personal_phone: '9998887777',
      status: 'current',
      date_requested: 'March 12, 2020 at 9:09:34 AM UTC-4',
      date_approved: 'March 13, 2020 at 10:15:34 PM UTC-4',
      date_denied: '',
      date_deactivated: 'March 12, 2020 at 9:09:34 AM UTC-4',
    };
    FSettingsMethods().toggleAccountActive(db, user, cfInfo , (time) => expect(time).toEqual(''));
    
  });
  
  
})