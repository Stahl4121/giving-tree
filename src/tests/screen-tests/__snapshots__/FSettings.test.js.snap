// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableData basics EditItem renders correctly 1`] = `ShallowWrapper {}`;

exports[`EditableData basics Renders correctly 1`] = `ShallowWrapper {}`;

exports[`FSettings Renders correctly 1`] = `ShallowWrapper {}`;

exports[`NoneditableData Renders correctly 1`] = `ShallowWrapper {}`;

exports[`NoneditableData StaticItem renders correctly 1`] = `ShallowWrapper {}`;
